src/components/Commands.tsx

```diff
@@ -48,8 +41,63 @@ const Command = (props: any) => {
 }
 
 const Commands = () => {
+  const [loading, setLoading] = useState(false)
+  const [isAsanaTaskFormVisible, setIsAsanaTaskFormVisible] = useState(false);
+  const [asanaFormRef, setAsanaFormRef] = useState()
+  const [credentials] = useAsanaCredentials();
+  const { createTask } = useCreateAsanaTask();
+
+  const isAsanaAuthorized = !!credentials.access_token
+  const commands = [{
+    name: 'Create Task',
+    icon: sunsamaLogo,
+    command: () => message.success('Creating a task is coming soon')
+  }, {
+    name: isAsanaAuthorized ? 'Create Asana Task' : 'Connect to Asana',
+    icon: asanaLogo,
+    command: () => {
+      if (!isAsanaAuthorized) {
+        window.location.replace(createClient().app.asanaAuthorizeUrl())
+      } else {
+        setIsAsanaTaskFormVisible(true)
+      }
+    }
+  }]
+
+  const handleAsanaTaskCreateCancel = () => {
+    const { form } = asanaFormRef.props;
+    form.resetFields();
+    setIsAsanaTaskFormVisible(false)
+  }
+
+  const handleAsanaTaskCreate = async () => {
+    setLoading(true)
+    const { form } = asanaFormRef.props;
+    form.validateFields(async (err: Error | null, values: any) => {
+      if (err) {
+        message.error(err.message)
+        setLoading(false)
+        return;
+      }
+
+      await createTask({
+        description: values.title
+      })
```

Comment:
It's likely that customers would want the task creation experience to feel fast. 

Instead of waiting for the task to create here it would be a better experience to kickoff the task creation process and show an alert if it were to fail. 

It might even be nice to show the "loading" dialog in the Asana panel instead of the in the task creation form so that the user can return to whatever they were doing before and have the feedback that the app was writing something to Asana.

Reply:
That is a good point.

I thought about going through the process of adding in error recovery and cleanup with a fake it first approach but I found the current lag to be reasonable. I agree with you though that moving the loading to the asana task list would probably be a smoother experience. I was thinking a custom CardLoading component with some entry animations like these ones https://cssanimation.rocks/list-items/ might be a nice touch.

Thanks for the feedback in a production environment I would definitely implement these changes.